tensorrt_llm/_torch/pyexecutor/sampler/sampler_features.py

```
@@ -0,0 +1,752 @@
+    # Source indices for bias application
+    bias_gather_indices: list[int] = []
+
+    # Collect bias information
+    #
+    # NB: the mask indexes rows of 'logits', not requests: a request contributes
+    #     'steps' consecutive rows (> 1 under speculative decoding), so the offset
+    #     has to accumulate 'steps' rather than track the request index. It must
+    #     stay in step with 'bias_gather_indices', which is built in row order.
+    req_bias = None
+    row_offset = 0
+    for req, steps in zip(requests, request_steps_list):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)